core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java

```diff
@@ -272,6 +413,17 @@ protected void startMasters(boolean isLeader) {
       mWorkerConfigChecker.reset();
       if (isLeader) {
         mSafeModeManager.notifyPrimaryMasterStarted();
+
+        //  The service that detects lost standby master nodes
+        mExecutorService.submit(new HeartbeatThread(
```

Comment:
if this becomes a standby, where does this executor service get shutdown or stopped?

Reply:
Add stopExecutorService in startMasters. When masters switch position, it will first shut down its executor service and then create a new executor service.
```
  protected void startMasters(boolean isLeader) {
    try {
      mWorkerConfigChecker.reset();
      if (isLeader) {
        stopExecutorService();
        mExecutorService = mExecutorServiceFactory.create();
        mExecutorService.submit....
      } else {
        stopExecutorService();
        mExecutorService = mExecutorServiceFactory.create();
        mExecutorService.submit...
    } ...
```